test/integration/targets/iam_role/tasks/main.yml

```
@@ -0,0 +1,1127 @@
+      - role_info.iam_roles[0].max_session_duration == 43200
+      - role_info.iam_roles[0].path == '/'
+      - '"permissions_boundary" not in role_info.iam_roles[0]'
+      - role_info.iam_roles[0].role_id == iam_role.iam_role.role_id
+      - role_info.iam_roles[0].role_name == test_role
+
+  # ===================================================================
+  # Inline Policy (test _info behaviour)
+
+  # XXX Not sure if it's a bug in Ansible or a "quirk" of AWS, but these two
+  # policies need to have at least different Sids or the second doesn't show
+  # up...
+  - name: 'Attach inline policy a'
+    iam_policy:
+      state: present
+      iam_type: 'role'
+      iam_name: '{{ test_role }}'
+      policy_name: 'inline-policy-a'
+      policy_json: '{{ lookup("file", "deny-all-a.json") }}'
+
+  - name: 'Attach inline policy b'
+    iam_policy:
+      state: present
+      iam_type: 'role'
+      iam_name: '{{ test_role }}'
+      policy_name: 'inline-policy-b'
+      policy_json: '{{ lookup("file", "deny-all-b.json") }}'
+
+  - name: 'iam_role_info after attaching inline policies (using iam_policy)'
+    iam_role_info:
+      name: '{{ test_role }}'
+    register: role_info
+  - assert:
+      that:
+      - role_info is succeeded
+      - role_info.iam_roles | length == 1
+      - 'role_info.iam_roles[0].arn.startswith("arn")'
+      - 'role_info.iam_roles[0].arn.endswith("role/" + test_role )'
+      - '"assume_role_policy_document" in role_info.iam_roles[0]'
+      - '"create_date" in role_info.iam_roles[0]'
+      - 'role_info.iam_roles[0].description == "Ansible Test Role (updated) {{ resource_prefix }}"'
+      - role_info.iam_roles[0].inline_policies | length == 2
+      - '"inline-policy-a" in role_info.iam_roles[0].inline_policies'
+      - '"inline-policy-b" in role_info.iam_roles[0].inline_policies'
+      - role_info.iam_roles[0].instance_profiles | length == 1
+      - role_info.iam_roles[0].instance_profiles[0].instance_profile_name == test_role
+      - 'role_info.iam_roles[0].instance_profiles[0].arn.startswith("arn")'
+      - 'role_info.iam_roles[0].instance_profiles[0].arn.endswith("instance-profile/" + test_role)'
+      - role_info.iam_roles[0].managed_policies | length == 1
+      - safe_managed_policy not in ( role_info | json_query("iam_roles[*].managed_policies[*].policy_name") | list | flatten )
+      - custom_policy_name in ( role_info | json_query("iam_roles[*].managed_policies[*].policy_name") | list | flatten )
+      - role_info.iam_roles[0].max_session_duration == 43200
+      - role_info.iam_roles[0].path == '/'
+      - '"permissions_boundary" not in role_info.iam_roles[0]'
+      - role_info.iam_roles[0].role_id == iam_role.iam_role.role_id
+      - role_info.iam_roles[0].role_name == test_role
+
+  # XXX iam_role fails to remove inline policies before deleting the role
+  - name: 'Attach inline policy a'
```

```suggestion
  - name: 'Detach inline policy a'
```
And the one below as well